cmd/mattermost/commands/config.go

```diff
@@ -224,6 +235,145 @@ func printMap(value reflect.Value, tabVal int) string {
 
 }
 
+func configSetCmdF(command *cobra.Command, args []string) error {
+	app, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+
+	defer app.Shutdown()
+
+	// args[0] -> holds the config setting that we want to change
+	// args[1:] -> the new value of the config setting
+	configSetting := args[0]
+	newVal := args[1:]
+
+	// Update the config
+
+	// first disable the watchers
+	app.DisableConfigWatch()
+
+	// create the function to update config
+	oldConfig := app.Config()
+	newConfig := app.Config()
+	f := updateConfigValue(configSetting, newVal, oldConfig, newConfig)
+
+	// update the config
+	app.UpdateConfig(f)
+
+	// make the changes persist
+	app.PersistConfig()
+
+	// reload config
+	app.ReloadConfig()
+
+	// Enable config watchers
+	app.EnableConfigWatch()
+
+	return nil
+}
+
+func updateConfigValue(configSetting string, newVal []string, oldConfig, newConfig *model.Config) func(*model.Config) {
+	return func(update *model.Config) {
+
+		// convert config to map[string]interface
+		configMap := configToMap(*oldConfig)
+
+		// iterate through the map and update the value or print an error and exit
+		err := updateMap(configMap, strings.Split(configSetting, "."), newVal)
+		if err != nil {
+			fmt.Printf("%s\n", err)
+			os.Exit(1)
+		}
+
+		// convert map to json
+		bs, err := json.Marshal(configMap)
+		if err != nil {
+			fmt.Printf("Error while marshalling map to json %s\n", err)
+			os.Exit(1)
+		}
+
+		// convert json to struct
+		err = json.Unmarshal(bs, newConfig)
+		if err != nil {
+			fmt.Printf("Error while unmarshalling json to struct %s\n", err)
+			os.Exit(1)
+		}
+
+		*update = *newConfig
+
+	}
+}
+
+func updateMap(configMap map[string]interface{}, configSettings []string, newVal []string) error {
+	res, ok := configMap[configSettings[0]]
+	if !ok {
+		return fmt.Errorf("config setting is not in the file")
+	}
+
+	value := reflect.ValueOf(res)
+
+	switch value.Kind() {
```

Comment:
Does this work for the config settings that are pointers? I'm not sure how the `value.Kind()` works for those